common/lib/xmodule/xmodule/html_module.py

```diff
@@ -253,6 +254,24 @@ def non_editable_metadata_fields(self):
         non_editable_fields.append(HtmlDescriptor.use_latex_compiler)
         return non_editable_fields
 
+    def index_dictionary(self):
+        xblock_body = super(HtmlDescriptor, self).index_dictionary()
+        html_content = re.sub(r"(\s|&nbsp;|//)+", " ", html_to_text(self.data))
+        html_content = re.sub(r"<!\[CDATA\[.*\]\]>", "", html_content)
+        html_content = re.sub(r"<!--.*-->", "", html_content)
+        html_body = {
+            "html_content": html_content,
+            "display_name": self.display_name,
+        }
+        if "content" in xblock_body:
+            xblock_body["content"].update(html_body)
+        else:
+            xblock_body.update({"content": html_body})
```

Comment:
Don't do `dict.update()` with only one key-value pair -- it's silly and over-engineered. You can rewrite this line as:
```python
xblock_body["content"] = html_body
```